test/extended/router/gatewayapicontroller.go

```diff
@@ -0,0 +1,294 @@
+package router
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	gatewayapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+	gatewayapiclientset "sigs.k8s.io/gateway-api/pkg/client/clientset/versioned"
+)
+
+var _ = g.Describe("[sig-network][OCPFeatureGate:GatewayAPIController][Feature:Router][apigroup:gateway.networking.k8s.io]", func() {
+	defer g.GinkgoRecover()
+	var (
+		oc = exutil.NewCLIWithPodSecurityLevel("gatewayapi-controller", admissionapi.LevelBaseline)
+	)
+	const (
+		// The expected OSSM subscription name.
+		expectedSubscriptionName = "servicemeshoperator3"
+		// The expected OSSM operator namespace.
+		expectedSubscriptionNamespace = "openshift-operators"
+		// The gatewayclass name used to create ossm and other gateway api resources.
+		gatewayClassName = "openshift-default"
+		// gatewayClassControllerName is the name that must be used to create a supported gatewayClass.
+		gatewayClassControllerName = "openshift.io/gateway-controller/v1"
+		// The gateway name used to create gateway resource.
+		gatewayName = "openshift-gateway"
+	)
+	g.BeforeEach(func() {
+		// create the default gatewayClass and confirm that it is accepted
+		gwapiClient := gatewayapiclientset.NewForConfigOrDie(oc.AdminConfig())
+
+		gatewayClass := buildGatewayClass(gatewayClassName, gatewayClassControllerName)
+		gwc, err := gwapiClient.GatewayV1().GatewayClasses().Create(context.TODO(), gatewayClass, metav1.CreateOptions{})
+		if err != nil {
+			e2e.Logf("Gateway Class %s already exists, or has failed to be created, checking its status", gwc.Name)
+		}
+
+		errCheck := checkGatewayClass(oc, gatewayClassName)
+		o.Expect(errCheck).NotTo(o.HaveOccurred())
+		e2e.Logf("GatewayClass %s successfully installed and accepted!", gwc.Name)
+
+	})
+
+	g.Describe("Verify Gateway API controller resources are created", func() {
+		g.It("and ensure OSSM related resources are created", func() {
```

Comment:
Please remove Ishmam tests and just keep your own tests in this PR.

Reply:
done